### PR TITLE
feat: authentication

### PR DESCRIPTION
**Pull request by pankaj-prog**
Added Sign in and SIgn out.
 Local storage added to persist the login information

### CODE REVIEW

**Review by iHarryD — APPROVED** [review]
- Code is exceedingly well written and well structured.
- Would recommend using a bit more descriptive CSS variable names.
- You don't need to specify the unit for zero.

File: src/context/authContext.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const AuthProvider = ({ children }) => {
+  const [user, setUser] = useState(null);
+  const [encodedToken, setEncodedToken] = useState(
+    localStorage.getItem("encodedToken")
+  );
+
+  return (
+    <AuthContext.Provider
+      value={{
+        user: user,
```
Comment (bhtibrewal) [inline src/context/authContext.jsx]: Suggestion: Use shortcircuiting here

File: src/pages/auth/utils/authHandlers.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+export const signUpHandler = async (
+  formState,
+  setFormState,
+  setUser,
+  setEncodedToken,
+  navigate
+) => {
+  // update form validation states
+  if (!validateName(formState.name)) {
+    setFormState((formState) => ({ ...formState, isNameValid: false }));
```
Comment (bhtibrewal) [inline src/pages/auth/utils/authHandlers.js:20]: Suggestion: Destructure formstate

File: src/pages/auth/utils/formInputUtils.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export const validatePassword = (password) => {
+  if (password.match(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,20}$/)) {
+    return true;
+  }
+  return false;
+};
+
+// accept alphabets and only space character between words, should accept min 2 chars and max of 30 chars
+export const validateName = (name) => {
+  if (/^[a-zA-Z ]{2,30}$/.test(name)) {
```
Comment (bhtibrewal) [inline src/pages/auth/utils/formInputUtils.js:18]: Return this itself

**Review by bhtibrewal — APPROVED** [review]
Good code just some small suggestions, after that good to merge

**Pull request merged**